File: backend/src/MyDatingProfileHelper.Backend.Api/DTOs/SaveGeneratedProfileHistoryRequest.cs
csharp
// Пространство имен для DTO API слоя
namespace MyDatingProfileHelper.Backend.Api.DTOs
{
    /// <summary>
    /// DTO для запроса сохранения истории сгенерированной анкеты.
    /// Используется для получения данных от клиента.
    /// </summary>
    public class SaveGeneratedProfileHistoryRequest
    {
        /// <summary>
        /// Сгенерированный текст анкеты.
        /// </summary>
        public string GeneratedText { get; set; }

        /// <summary>
        /// Входные параметры, использованные для генерации (в формате JSON или строки).
        /// </summary>
        public string? InputParameters { get; set; }
    }
}

File: app/src/main/java/com/jounik_projects/mydatingprofilehelper/data/network/model/SaveGeneratedProfileHistoryRequest.kt
package com.jounik_projects.mydatingprofilehelper.data.network.model

/**
 * Модель данных для запроса сохранения истории сгенерированного профиля на бэкенде.
 */
data class SaveGeneratedProfileHistoryRequest(
    val generatedText: String, // Сгенерированный текст анкеты
    val inputParameters: String? // Входные параметры, использованные для генерации (могут быть в формате JSON)
)
package com.jounik_projects.mydatingprofilehelper.data.network.model

/**
 * Модель данных для запроса сохранения истории сгенерированного профиля на бэкенде.
 */
data class SaveGeneratedProfileHistoryRequest(
    val generatedText: String, // Сгенерированный текст анкеты
    val inputParameters: String? // Входные параметры, использованные для генерации (могут быть в формате JSON)
)